sdk/keyvault/keyvault-keys/src/cryptographyClient.ts

```diff
@@ -125,6 +130,13 @@ export class CryptographyClient {
         value: key
       };
     }
+
+    if (pipelineOptions.remoteOnly) {
+      if (this.key.kind === "JsonWebKey") {
+        throw new Error("The remoteOnly flag cannot be used with a local JsonWebKey");
```

Comment:
This is where it gets complicated. Technically, if the JsonWebKey.id is a vault URI, it could be remote only - the rest of the JWK is superfluous then.